servers/physics_2d/godot_space_2d.cpp

```diff
@@ -570,349 +564,353 @@ bool GodotSpace2D::test_body_motion(GodotBody2D *p_body, const PhysicsServer2D::
 		return false;
 	}
 
-	real_t margin = MAX(p_parameters.margin, TEST_MOTION_MARGIN_MIN_VALUE);
+	bool did_collide = false;
+
+	tbmd.margin = MAX(p_parameters.margin, TEST_MOTION_MARGIN_MIN_VALUE);
 
 	// Undo the currently transform the physics server is aware of and apply the provided one
-	body_aabb = p_parameters.from.xform(p_body->get_inv_transform().xform(body_aabb));
-	body_aabb = body_aabb.grow(margin);
+	tbmd.body_aabb = p_parameters.from.xform(p_body->get_inv_transform().xform(tbmd.body_aabb));
+	tbmd.body_aabb = tbmd.body_aabb.grow(tbmd.margin);
 
-	static const int max_excluded_shape_pairs = 32;
-	ExcludedShapeSW excluded_shape_pairs[max_excluded_shape_pairs];
-	int excluded_shape_pair_count = 0;
+	tbmd.min_contact_depth = tbmd.margin * TEST_MOTION_MIN_CONTACT_DEPTH_FACTOR;
 
-	real_t min_contact_depth = margin * TEST_MOTION_MIN_CONTACT_DEPTH_FACTOR;
+	tbmd.motion_length = p_parameters.motion.length();
+	tbmd.motion_normal = p_parameters.motion / tbmd.motion_length;
 
-	real_t motion_length = p_parameters.motion.length();
-	Vector2 motion_normal = p_parameters.motion / motion_length;
+	tbmd.body_transform = p_parameters.from;
 
-	Transform2D body_transform = p_parameters.from;
+	free_stuck_body(p_body, p_parameters, &tbmd);
 
-	bool recovered = false;
+	attempt_body_motion(p_body, p_parameters, &tbmd);
 
-	{
-		//STEP 1, FREE BODY IF STUCK
+	did_collide = retrieve_body_motion_data(p_body, p_parameters, r_result, &tbmd);
 
-		const int max_results = 32;
-		int recover_attempts = 4;
-		Vector2 sr[max_results * 2];
-		real_t priorities[max_results];
+	return did_collide;
+}
 
-		do {
-			GodotPhysicsServer2D::CollCbkData cbk;
-			cbk.max = max_results;
-			cbk.amount = 0;
-			cbk.passed = 0;
-			cbk.ptr = sr;
-			cbk.invalid_by_dir = 0;
-			excluded_shape_pair_count = 0; //last step is the one valid
+void GodotSpace2D::free_stuck_body(GodotBody2D *p_body, const PhysicsServer2D::MotionParameters &p_parameters, TestBodyMotionData *p_tbmd) {
+	// STEP 1
 
-			GodotPhysicsServer2D::CollCbkData *cbkptr = &cbk;
-			GodotCollisionSolver2D::CallbackResult cbkres = GodotPhysicsServer2D::_shape_col_cbk;
-			int priority_amount = 0;
+	const int max_results = 32;
+	int recover_attempts = 4;
+	Vector2 sr[max_results * 2];
+	real_t priorities[max_results];
 
-			bool collided = false;
+	do {
+		GodotPhysicsServer2D::CollCbkData cbk;
+		cbk.max = max_results;
+		cbk.amount = 0;
+		cbk.passed = 0;
+		cbk.ptr = sr;
+		cbk.invalid_by_dir = 0;
+		p_tbmd->excluded_shape_pair_count = 0; //last step is the one valid
```

Comment:
```suggestion
		p_tbmd->excluded_shape_pair_count = 0; // Last step is the one valid.
```
[Comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide) nitpick, including code rearranged or copied